File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    answer = 0
    # 가로와 세로 값들 저장 리스트
    garo=[]
    sero=[]
    
    # 명함을 돌릴 수 있으므로 큰 값을 전부 가로에
    # 가로보다 세로가 크면 값을 바꿈
    # 그 후 리스트에 추가
    for i in range(len(sizes)):
        if sizes[i][0]<sizes[i][1]:
            tmp=sizes[i][0]
            sizes[i][0]=sizes[i][1]
            sizes[i][1]=tmp
        garo.append(sizes[i][0])
        sero.append(sizes[i][1])
    
    # 가로의 최대값과 세로의 최대값을 곱함
    answer=max(garo)*max(sero)
    return answer

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.*;

class Solution {
    public int solution(int[][] sizes) {
        int len=sizes.length;
        int answer = 0;
        
        // 명함을 돌릴 수 있으므로 큰 값을 가로에 작은 값을 세로에
        ArrayList<Integer> garo = new ArrayList<>();
        ArrayList<Integer> sero = new ArrayList<>();
        
        for (int i=0; i<len; i++) {
            if (sizes[i][0]>=sizes[i][1]){
                garo.add(sizes[i][0]);
                sero.add(sizes[i][1]);
            }
            else{
                garo.add(sizes[i][1]);
                sero.add(sizes[i][0]);
            }
        }
        answer=Collections.max(garo)* Collections.max(sero);
        return answer;
    }
}